.github/actions/build-onedir-salt/action.yml

```diff
@@ -0,0 +1,106 @@
+---
+name: build-onedir-salt
+description: Build Onedir Package
+inputs:
+  platform:
+    required: true
+    type: string
+    description: The platform to build
+  arch:
+    required: true
+    type: string
+    description: The platform arch to build
+  package-name:
+    required: false
+    type: string
+    description: The onedir package name to create
+    default: salt
+  cache-seed:
+    required: true
+    type: string
+    description: Seed used to invalidate caches
+
+runs:
+  using: composite
+
+  steps:
+
+    - name: Download Cached Deps Onedir Package Directory
+      id: onedir-bare-cache
+      uses: actions/cache@v3
+      with:
+        path: artifacts/${{ inputs.package-name }}
+        key: ${{ inputs.cache-seed }}|relenv|${{ env.RELENV_VERSION }}|deps|${{ inputs.platform }}|${{ inputs.arch }}|${{ inputs.package-name }}|${{ hashFiles(format('{0}/.relenv/**/*.xz', github.workspace), 'requirements/static/pkg/*/*.txt') }}
+
+    - name: Download Source Tarball
+      uses: actions/download-artifact@v3
+      with:
+        name: salt-${{ env.SALT_VERSION }}.tar.gz
+
+    - name: Install Salt Into Onedir
+      if: ${{ inputs.platform != 'windows' }}
+      env:
+        PIP_INDEX_URL: https://pypi-proxy.saltstack.net/root/local/+simple/
+        PIP_EXTRA_INDEX_URL: https://pypi.org/simple
+        USE_STATIC_REQUIREMENTS: "1"
+        RELENV_PIP_DIR: "1"
+      shell: bash
+      run: |
+        artifacts/${{ inputs.package-name }}/bin/python3 -m pip install salt-${{ env.SALT_VERSION }}.tar.gz
+        if [ ${{ inputs.platform }} == "darwin" ]; then
+          pkg/macos/prep_salt.sh --build-dir ./artifacts/${{ inputs.package-name }}
+          rm -rf ./artifacts/${{ inputs.package-name }}/opt
+          rm -rf ./artifacts/${{ inputs.package-name }}/etc
+          rm -rf ./artifacts/${{ inputs.package-name }}/Library
+        fi
+
+    - name: Install Salt Into Onedir (Windows)
+      if: ${{ inputs.platform == 'windows' }}
+      env:
+        PIP_INDEX_URL: https://pypi-proxy.saltstack.net/root/local/+simple/
+        PIP_EXTRA_INDEX_URL: https://pypi.org/simple
+        USE_STATIC_REQUIREMENTS: "1"
+      shell: powershell
+      run: |
+        # install salt
+        pkg\windows\install_salt.cmd -BuildDir ".\artifacts\${{ inputs.package-name }}" -CICD -SourceTarball salt-${{ env.SALT_VERSION }}.tar.gz
+        # prep salt
+        pkg\windows\prep_salt.cmd -BuildDir ".\artifacts\${{ inputs.package-name }}" -CICD
```

Comment:
Aren't we doing this in the `pre-archive-cleanup` now?

Reply:
Yes, but I didn't want to completely break your scripts, since you don't use that file on your scripts....